mautic/mautic
# Refactor Form Submit Actions

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Automated tests included? | N
| Related user documentation PR URL | N/A
| Related developer documentation PR URL | N/A
| Issues addressed (#s or URLs) | #8066 
| BC breaks? | Y
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Form submit actions had supports for proper events added a while back. This PR leverages the form submit event by removing callback support, and refactoring the callbacks away from usage of MauticFactory.

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Ensure all form submit actions continue to function as expected.

#### List backwards compatibility breaks:
1. LeadEvents::FORM_SUBMIT_REMOVE_DO_NOT_CONTACT event is removed. Listen for the LeadEvents::ON_EXECUTE_SUBMIT_ACTION instead, and check the context of `lead.remove_do_not_contact`
